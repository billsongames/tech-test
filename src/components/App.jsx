import { React, useState } from "react";
import Search from "./Search";
import SearchResults from "./SearchResults";

function App() {
  const [searchResults, setSearchResults] = useState([])
  const [initialise, setInitialise] = useState(false)

  return (
    <div className="app">
      <Search setSearchResults={setSearchResults} setInitialise={setInitialise} />
      {initialise && < SearchResults results={searchResults} />}
    </div>
  );
}

export default App;
