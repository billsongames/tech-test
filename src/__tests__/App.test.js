import { render } from "@testing-library/react";
import App from "../components/App";

describe('App', () => {
  test("app renders correctly", () => {
    const { asFragment } = render(<App />)
    
    expect(asFragment()).toMatchSnapshot()
  })
})
