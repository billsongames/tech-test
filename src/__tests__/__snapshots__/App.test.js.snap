// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App app renders correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div>
      <div
        class="header"
      >
        <div
          class="nasa-logo"
        >
          <img
            alt="nasa logo"
            class="nasa-logo__image"
            src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
          />
        </div>
      </div>
      <div>
        <form
          class="input-search"
        >
          <input
            class="search-box"
            placeholder="Search..."
            type="text"
          />
          <button
            class="search-button"
            type="submit"
          >
            Launch
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
